tests/e2e/paths/search/granules/granules.spec.js

```diff
@@ -0,0 +1,1197 @@
+    test('loads with the online only checkbox checked', async ({ page }) => {
+      const conceptId = 'C1214470488-ASF'
+      const cmrHits = 1059331
+
+      await page.route('**/search/granules.json', (route) => {
+        const request = route.request()
+        const body = request.postData()
+        expect(body).toBe('echo_collection_id=C1214470488-ASF&online_only=true&page_num=1&page_size=20')
+        route.fulfill({
+          body: JSON.stringify(onlineOnlyGranulesBody),
+          headers: {
+            ...commonGranulesHeaders,
+            'access-control-expose-headers': 'cmr-hits',
+            'cmr-hits': cmrHits
+          }
+        })
+      })
+
+      await page.route('**/search/granules/timeline', (route) => {
+        const request = route.request()
+        const body = request.postData()
+        if (body) {
+          expect(body).toBe('end_date=2023-12-01T00:00:00.000Z&interval=day&start_date=2018-12-01T00:00:00.000Z&concept_id[]=C1214470488-ASF')
+        }
+
+        route.fulfill({
+          body: JSON.stringify(onlineOnlyTimelineBody),
+          headers: onlineOnlyTimelineHeaders
+        })
+      })
+
+      await page.route('**/api', (route) => {
+        const requestBody = JSON.parse(route.request().postData())
+        const expectedBody = JSON.parse(graphQlGetCollection(conceptId))
+
+        expect(requestBody).toEqual(expectedBody)
+        route.fulfill({
+          body: JSON.stringify(onlineOnlyGraphQlBody),
+          headers: onlineOnlyGraphQlHeaders
+        })
+      })
+
+      await page.goto('/search/granules?p=C1214470488-ASF&pg[0][oo]=true')
+
+      // Ensure the correct number of results were loaded
+      await testResultsSize(page, cmrHits)
+
+      // Checkboxes are checked correctly
+      await expect(page.getByTestId('granule-filters__browse-only')).not.toBeChecked()
+      await expect(page.getByTestId('granule-filters__online-only')).toBeChecked()
+    })
+  })
+
+  test.describe('When the path is loaded with orbit number parameters', () => {
+    test('loads with the orbit number fields populated', async ({ page }) => {
+      const conceptId = 'C1214470488-ASF'
+      const cmrHits = 227
+
+      await page.route('**/search/granules.json', (route) => {
+        const request = route.request()
+        const body = request.postData()
+        expect(body).toBe('echo_collection_id=C1214470488-ASF&orbit_number[min]=30000&orbit_number[max]=30005&page_num=1&page_size=20')
+        route.fulfill({
+          body: JSON.stringify(orbitNumberGranulesBody),
+          headers: {
+            ...commonGranulesHeaders,
+            'access-control-expose-headers': 'cmr-hits',
+            'cmr-hits': cmrHits
+          }
+        })
+      })
+
+      await page.route('**/search/granules/timeline', (route) => {
+        const request = route.request()
+        const body = request.postData()
+        if (body) {
```

Comment:
It seems kind of odd to have an if block in a test. I see this in a few spots. Why would body not exist when running this test?

Reply:
I wondered the same thing, but this was the functionality in the previous test file I converted it from. I did console log both parts of the conditional in a few of these tests to see if they were both getting hit and it seemed like both were getting hit at times.